enclave/core/sgx/exception.c

```diff
@@ -385,20 +395,28 @@ void oe_virtual_exception_dispatcher(
         *arg_out = OE_EXCEPTION_CONTINUE_SEARCH;
         return;
     }
-    sgx_ssa_gpr_t* ssa_gpr =
-        (sgx_ssa_gpr_t*)(((uint8_t*)ssa_info.base_address) + ssa_info.frame_byte_size - OE_SGX_GPR_BYTE_SIZE);
+    uint64_t gprsgx_offset = (uint64_t)ssa_info.base_address +
+                             ssa_info.frame_byte_size - OE_SGX_GPR_BYTE_SIZE;
+    sgx_ssa_gpr_t* ssa_gpr = (sgx_ssa_gpr_t*)gprsgx_offset;
+    bool is_exit_info_valid = ssa_gpr->exit_info.as_fields.valid;
 
-    if (!ssa_gpr->exit_info.as_fields.valid)
+    if (!is_exit_info_valid)
     {
         // Consider manually provided exception information for SGX1
-        if (oe_exception_record)
+        // Only allow the simulated #PF in debug mode
+        if (is_enclave_debug_allowed() && oe_exception_record)
         {
             td->exception_address = ssa_gpr->rip;
 
             /* TODO PRP: We need to read the exception information from
```

Comment:
This TODO comment is obsolete, with the design of "only allow the simulated #PF in debug mode".

Reply:
removed the comment